libraries/abstractions/common_io/include/iot_spi.h

```diff
@@ -244,94 +298,117 @@ int32_t iot_spi_write_sync( IotSPIHandle_t const pxSPIPeripheral,
  */
 int32_t iot_spi_write_async( IotSPIHandle_t const pxSPIPeripheral,
                              uint8_t * const pvBuffer,
                              size_t xBytes );
 
 /**
- * @brief The SPI master starts a synchronous transfer between master and the
- * slave.
+ * @brief The SPI master starts a synchronous transfer between master and the slave.
  *
- * The application should call this function to start the transfer with number
- * of bytes specified. The SPI driver fills in xBytes in RxBuffer and reads
- * xBytes in TxBuffer.
+ * This function attempts to read/write certain number of bytes from/to two pre-allocated buffers at the same time, in synchronous way.
+ * Partial read/write might happen, e.g. slave device unable to receive more data.
+ * And the number of bytes that have been actually written can be obtained by calling iot_spi_ioctl.
  *
  * @param[in] pxSPIPeripheral The SPI peripheral handle returned in open() call.
  * @param[in] pvTxBuffer The buffer to store the received data.
  * @param[out] pvRxBuffer The buffer with data to transmit.
  * @param[in] xBytes The number of bytes to transfer.
  *
- * @return SUCCESS = IOT_SPI_SUCCESS,
- *          FAILED = IOT_SPI_INVALID_VALUE,
- *          SPI_BUSY = IOT_SPI_BUS_BUSY,
- *          MASTER_UNINITALIZED = IOT_SPI_MASTER_NOT_INITIALIZED,
- *          Slave not Selected = IOT_SPI_SLAVE_NOT_SELECTED
+ * @return
+ * - IOT_SPI_SUCCESS, on success (all the requested bytes have been read/written)
+ * - IOT_SPI_INVALID_VALUE, if
+ *     - pxSPIPeripheral is NULL
+ *     - pxSPIPeripheral is not opened yet
+ *     - pucBuffer is NULL
+ *     - xBytes is 0
+ * - IOT_SPI_TRANSFER_ERROR, if there is some unknown driver error.
+ * - IOT_SPI_BUSY, if the bus is busy which means there is an ongoing transaction.
  */
 int32_t iot_spi_transfer_sync( IotSPIHandle_t const pxSPIPeripheral,
                                uint8_t * const pvTxBuffer,
                                uint8_t * const pvRxBuffer,
                                size_t xBytes );
 
 /**
- * @brief The SPI master starts a asynchronous transfer between master and the
- * slave.
+ * @brief The SPI master starts a asynchronous transfer between master and the slave.
+ *
+ * This function attempts to read/write certain number of bytes from/to two pre-allocated buffers at the same time, in asynchronous way.
+ * It returns immediately when the operation is started and the status can be check by calling iot_spi_ioctl.
+ * Once the operation completes, no matter successful or failed, the user callback will be invoked.
  *
- * The application should call this function to start the asynchronous transfer
- * with number of bytes specified. The SPI driver fills in xBytes in RxBuffer
- * and reads xBytes in TxBuffer when the SPI bus is free.
+ * Partial read/write might happen. And the number of bytes that have been actually read can be obtained by calling iot_spi_ioctl.
```

Comment:
Similar here. Lets be clear that on partial read/writes callback is not called. (though you said "Once operation completes, no matter successful or failed, the user callback will be invoked")